blocks/subscriptions-block/components/ActionDialog/index.jsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import PropTypes from '@arc-fusion/prop-types';
+
+import {
+  Button,
+  BUTTON_SIZES,
+  BUTTON_STYLES,
+  BUTTON_TYPES,
+  PrimaryFont,
+} from '@wpmedia/shared-styles';
+
+import './styles.scss';
+
+const ActionDialog = ({
+  actionText,
+  actionUrl,
+  reasonPromptText,
+  headlineText,
+  linkAriaLabel,
+  linkText,
+  linkPromptText,
+  linkUrl,
+  subHeadlineText,
+}) => (
+  <PrimaryFont as="div" className="xpmedia-action-dialog">
+    {reasonPromptText
+      ? (
+        <div
+          className="xpmedia-action-dialog-reason-prompt"
+          dangerouslySetInnerHTML={{ __html: reasonPromptText }}
+        />
+      )
+      : null}
+    <div className="xpmedia-action-dialog-link-prompt">
+      {linkPromptText
+        ? (
+          <span
+            className="xpmedia-action-dialog-link-prompt-pre-link"
+            dangerouslySetInnerHTML={{ __html: linkPromptText }}
+          />
+        )
+        : null}
+      <a
+        className="xpmedia-action-dialog-link-prompt-link"
+        href={linkUrl}
+        {...(linkAriaLabel ? { 'aria-label': linkAriaLabel } : {})}
+      >
+        {linkText}
+      </a>
+    </div>
+    {headlineText
+      ? (
+        <h2
+          className="xpmedia-action-dialog-headline"
+          dangerouslySetInnerHTML={{ __html: headlineText }}
+        />
+      )
+      : null}
+    {subHeadlineText
+      ? (
+        <h3
+          className="xpmedia-action-dialog-subheadline"
+          dangerouslySetInnerHTML={{ __html: subHeadlineText }}
+        />
+      )
+      : null}
+    {actionUrl && actionText
+      ? (
+        <Button
+          as="a"
+          buttonSize={BUTTON_SIZES.LARGE}
+          buttonStyle={BUTTON_STYLES.FILLED}
+          buttonType={BUTTON_TYPES.LABEL_ONLY}
+          href={actionUrl}
+          text={actionText}
+        />
+      )
+      : null}
+  </PrimaryFont>
+);
+
+ActionDialog.propTypes = {
+  actionText: PropTypes.string,
+  actionUrl: PropTypes.string,
+  reasonPromptText: PropTypes.string,
+  headlineText: PropTypes.string,
+  linkText: PropTypes.string.isRequired,
```

Comment:
linkText is required?

Reply:
Yeah, in the description of the log, this and the url is required... now, being a "generic" form might exclude that, so I would not be opposed to "optioning" these.
![image](https://user-images.githubusercontent.com/2287238/134066259-531ab0bd-01b3-4429-86d8-a806813bfcc0.png)

Reply:
Seems odd they are the only two items required - I must of missed that in grooming

Reply:
Brought this up to Jenae and she reminded me that there will be no way to exit this dialog for the user if they have an account so that's why it was marked as required.  Wether that's appropriate or not is undecided.  I think it could easily be a "soft" requirement.